src/Cache/Cache.php

```diff
@@ -113,6 +114,26 @@ class Cache
     protected static $_registry;
 
     /**
+     * Returns the Cache Registry instance used for creating and using cache adapters.
+     * Also allows for injecting of a new registry instance.
+     *
+     * @param \Cake\Core\ObjectRegistry $registry Injectable registry object.
+     * @return \Cake\Cache\CacheRegistry
+     */
+    public static function registry(ObjectRegistry $registry = null)
```

Comment:
The `@return` tag is lying if you can also set `ObjectRegistry` instance. Is any actual use case for custom registry? Also if indeed required custom registry can extend `CacheRegistry`.